src/components/Autosizer.tsx

```diff
@@ -0,0 +1,43 @@
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+import React, { useRef, useEffect, useState } from 'react';
+import { css } from '@emotion/css';
+
+interface ChildrenArgs {
+  width: number;
+  height: number;
+}
+
+interface Props {
+  children: (dimensions: ChildrenArgs) => JSX.Element;
+}
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  fillContainer: css`
+    display: flex;
+    justify-content: center;
+    max-width: 1300px;
+    margin-left: auto;
+    margin-right: auto;
+    width: 100%;
+  `,
+});
+
+export const Autosizer = ({ children }: Props) => {
+  const el = useRef<HTMLDivElement>(null);
+  const styles = useStyles2(getStyles);
+  const [size, setSize] = useState({ width: 0, height: 0 });
+
+  useEffect(() => {
+    const rect = el.current?.getBoundingClientRect();
+    if (rect) {
+      setSize({ width: rect.width, height: rect.height });
+    }
+  }, [el]);
```

Comment:
Shouldn't this be `el.current`?

Reply:
I had thought so too! When I do that, though, I get this eslint warning:

```
React Hook useEffect has an unnecessary dependency: 'el.current'. Either exclude it or remove the dependency array. Mutable values like 'el.current' aren't valid dependencies because mutating them doesn't re-render the component.
```

Reply:
Usually I would destructure the `el` object, get the current and use it directly, but this is an absolute nitpick.